rc/tools/go/gopls.kak

```diff
@@ -0,0 +1,98 @@
+# gopls.kak: gopls bindings for kakoune
+
+define-command -docstring "Enter gopls mode.
+Bindings for gopls in kakoune.
+
+Best used with mapping like:
+	map global normal '<whatever>' ': gopls<ret>'
+" \
+gopls %{ require-module gopls; evaluate-commands 'enter-user-mode gopls' }
+
+provide-module gopls %§
+
+evaluate-commands %sh{
+    if ! command -v gopls > /dev/null 2>&1; then
+        echo "fail Please install gopls or add to PATH!"
+    fi
+}
+
+try %{ declare-user-mode gopls }
+
+# Auto-format and auto-imports
+# ----------------------------
+
+map global gopls -docstring "format buffer" 'f' '<esc>: gopls-cmd format<ret>'
+map global gopls -docstring "fix buffer imports" 'i' '<esc>: gopls-cmd imports<ret>'
+
+declare-option -hidden str gopls_tmp_dir
+
+define-command -hidden -params 0 gopls-prepare %{
+    evaluate-commands %sh{
+        dir=$(mktemp -d "${TMPDIR:-/tmp}"/kak-gopls.XXXXXXXX)
+        printf %s\\n "set-option buffer gopls_tmp_dir ${dir}"
+    }
+}
+
+define-command -hidden -params 1 gopls-cmd %{
+    gopls-prepare
+    evaluate-commands %sh{
+        dir=${kak_opt_gopls_tmp_dir}
+        eval "gopls $1 -w ${kak_buffile} 2> ${dir}/stderr"
+        if [ $? -ne 0 ]; then
+            # show error messages in *debug* buffer
+            printf %s\\n "echo -debug '$(cat ${dir}/stderr)'"
```

Comment:
You could use the `%file{…}` expansion here.

Reply:
Sure, thanks.